pkg/cli/run.go

```diff
@@ -313,20 +313,18 @@ func (s *Run) update(ctx context.Context, c client.Client, imageSource imagesour
 		return nil, false, err
 	}
 
-	if !imageSource.IsImageSet() {
-		// If there is no image set, then lookup the existing app and use the image of the current app
-		imageSource = imageSource.WithImage(app.Status.AppImage.ID)
-	}
```

Comment:
This was the main source of the bug. We were taking `app.Status.AppImage.ID` (which could be a number of different things), setting that as the `--image` parameter quietly for `acorn update`, and then moving along.

For Docker Hub in particular this caused issues, because the ID would have `index.docker.io` as the prefix, even if the image Name didn't even specify the registry. This would cause the controller handler to re-resolve the image and potentially locate a new one even though the user never asked for the image to change.